lib/commands/misc/giverole.js

```diff
@@ -26,14 +26,38 @@ module.exports = class GiveRoleCommand extends Command {
 
 	async run(msg, args) {
     if (args.role === "member") {
-      const role = msg.guild.roles.find(r => r.name == 'Member')
-      if (!msg.member.roles.has(role)) {
+      const role = msg.guild.roles.find(r => r.name == 'Member');
+
+      if (!msg.member.roles.has(role.id)) {		// Needs to use role.id to properly check if the user has that role -Syn
         msg.member.addRole(role);
         const channel = this.client.channels.get(getChannel(GENERAL_ID));
         channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
       }
-    } else {
-      msg.reply(`Invalid role! Available options are: member`)
+
+			// Prevents this generalized code from allowing random people to self-assign the admin, moderator, or bae roles.
+    } else if (COLORS.includes(args.role)) {
+
+				const channel = this.client.channels.get(getChannel(REOL_CHAN_ID));
+				const role = msg.guild.roles.find(r => r.name == args.role);
+				const memberRole = msg.guild.roles.find(r => r.name == 'Member');
+
+				// Confirms that the role provided by the user exists. If it doesn't, the method returns after informing the user.
+				try {
+					let roleExists = msg.guild.roles.has(role.id);
+				} catch (err) {
+					msg.reply(`Invalid role! Available options can be found with r!roles`);
```

Comment:
minor: Since we're not using string template syntax, the backticks are redundant. Please use single quotes or double quotes for regular strings where appropriate